code/script.js

```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
     `
   })
 }
+// Aditional popup functionality  
+const customPopup = (message, multipleButtons, suspect) => {
+
+  popup.classList.add('custom-popup-on')
+ 
+  popupText.innerHTML = `
+  <h2>${message}</h2>`
+
+  if (multipleButtons) {
+    popupButtons.innerHTML = `
+    <button id="Ok" class="custom-button" onclick="onSelectConfirm('${suspect}')">Accept</button>
+    <button id="Cancel" class="custom-button" onclick="onSelectConfirm(false)">Cancel</button>
+    `
+  } else {
+    popupButtons.innerHTML = `
+    <button id="Ok" class="custom-button" onclick="closePopud()">Accept</button>
+     `
+  }
+
+}
+const closePopud = () => {
+  popup.classList.remove('custom-popup-on')
+}
+const onSelectConfirm = (suspect) => {
+
+  if (suspect) {
+    checkMyGuess(suspect)
+  }
+
+  closePopud()
+
+}
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  console.log(secret)
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  generateBoard()
+  setSecret()
+  //to restart the counter
+  counter.innerHTML = ""
+  guessCounter = 0
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
+
+  // variable created to assign the selected value
+  const selectedValue = questions.value
+
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: selectedValue,
+      category: category,
```

Comment:
you do not need to specify the value if you use the variable with the same name as a key, so `{category: category}` can be written as `{category}`.